.github/workflows/release.yml

```diff
@@ -8,36 +6,29 @@ on:
 
 jobs:
   pypi:
-    name: Publish to PyPI registry
+    name: release ${{ github.event.ref }}
     environment: release
     runs-on: ubuntu-22.04
+    permissions:
+      id-token: write
 
     env:
       FORCE_COLOR: 1
       PY_COLORS: 1
-      TOXENV: pkg
+      TOX_PARALLEL_NO_SPINNER: 1
 
     steps:
-      - name: Switch to using Python 3.10 by default
+      - name: Switch to using Python 3.12 by default
         uses: actions/setup-python@v5
         with:
-          python-version: 3.10
-
+          python-version: "3.12"
       - name: Install tox
         run: python3 -m pip install --user "tox>=4.0.0"
-
       - name: Check out src from Git
         uses: actions/checkout@v4
         with:
           fetch-depth: 0 # needed by setuptools-scm
-          submodules: true
-
       - name: Build dists
-        run: python -m tox
-
+        run: python -m tox -e pkg
```

Comment:
FWIW dists **MUST NOT** be built in a job that has OIDC access. It must be a separate job with reduced privileges. This is due to the possibility of privilege elevation though build dep poisoning that would compromise accesses beyond just the PyPI.